geps/gep-1867.md

```diff
@@ -0,0 +1,128 @@
+# GEP-1867: Per-Gateway Infrastructure
+
+* Status: Provisional
+* Issue: [#1876](https://github.com/kubernetes-sigs/gateway-api/issues/1876)
+
+## Overview
+
+`Gateway`s represent a piece of infrastructure implemented by cloud load balancers, in-cluster deployments, or other mechanisms.
+These often need vendor-specific configuration outside the scope of existing APIs (e.g. "size" or "version" of the infrastructure to provision).
+
+Today `GatewayClass.spec.parametersRef` is available to attach arbitrary configuration to a `GatewayClass`.
+
+This GEP will explain why that is not sufficient to meet common use cases, and introduce a new field - `infrastructure` - to address these cases.
+
+Related discussions:
+* [Support cluster-local Gateways](https://github.com/kubernetes-sigs/gateway-api/discussions/1247)
+* [Scaling Gateway Resources](https://github.com/kubernetes-sigs/gateway-api/discussions/1355)
+* [Manual deployments](https://github.com/kubernetes-sigs/gateway-api/issues/1687)
+* [Merging Gateways](https://github.com/kubernetes-sigs/gateway-api/pull/1863)
+* [In Cluster Gateway Deployments](https://github.com/kubernetes-sigs/gateway-api/pull/1757)
+
+## Goals
+
+* Provide the ability to configure arbitrary (implementation specific) attributes about a **specific Gateway**.
+* Provide the ability to configure a standardized set of attributes about a **specific Gateway**.
```

Comment:
These goals feel incomplete because you're also proposing standard infra-level config at the GatewayClass level but these only focus on individual Gateways.

Reply:
The implicit goal on the GEP is "Have consistency in the API". Providing `infrastructure` is purely to maintain consistency between the APIs; the entire goal of the GEP is to _not_ use GatewayClass.

Reply:
>the entire goal of the GEP is to not use GatewayClass

This seems problematic. Not to use GatewayClass _at all_ or just for this specific set of config?

Reply:
It is to be able to configure infrastructure options on a per-gateway basis. Its not to stop users from using GC, but they can already do that today. They cannot do it on a per-GW basis today, so this GEP aims to enable this.